components/suggest/src/store.rs

```diff
@@ -2051,6 +2077,15 @@ pub(crate) mod tests {
             .has_location_sign(false)
             .subject_exact_match(false)],
         );
+        assert_eq!(
+            store.fetch_suggestions(SuggestionQuery::yelp("best sp")),
+            vec![ramen_suggestion(
+                "best spicy ramen",
+                "https://www.yelp.com/search?find_desc=best+spicy+ramen"
+            )
+            .has_location_sign(false)
+            .subject_exact_match(false)],
+        );
```

Comment:
Could you add some queries like these please, where a modifier/subject is a substring of a query word?

```
"ramenabc"
"ramenabc xyz"
"best ramenabc"
"bestabc ra"
"bestabc ramen"
"bestabc ramen xyz"
```

Also different combinations of prefixes like these:

```
"best spi ram"
"bes ram"
"bes ramen"
```